ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -118,22 +139,30 @@ class MySqlNamedBlobDb implements NamedBlobDb {
   private static final String SELECT_FOR_SOFT_DELETE_QUERY =
       String.format("SELECT %s, %s, %s FROM %s WHERE %s FOR UPDATE", BLOB_ID, DELETED_TS, CURRENT_TIME, NAMED_BLOBS,
           PK_MATCH);
+  private static final String SELECT_FOR_SOFT_DELETE_QUERY_V2 =
+      String.format("SELECT %s, %s, %s, %s FROM %s WHERE %s AND %s ORDER BY %s DESC LIMIT 1 FOR UPDATE", BLOB_ID,
+          VERSION, DELETED_TS, CURRENT_TIME, NAMED_BLOBS_V2, PK_MATCH, STATE_MATCH, VERSION);
 
   /**
    * Soft delete a blob by setting the delete timestamp to the current time.
    */
   private static final String SOFT_DELETE_QUERY =
       String.format("UPDATE %s SET %s = ? WHERE %s", NAMED_BLOBS, DELETED_TS, PK_MATCH);
+  private static final String SOFT_DELETE_QUERY_V2 =
+      String.format("UPDATE %s SET %s = ? WHERE %s", NAMED_BLOBS_V2, DELETED_TS, PK_MATCH_VERSION);
 
   private final AccountService accountService;
   private final String localDatacenter;
   private final List<String> remoteDatacenters;
   private final RetryExecutor retryExecutor;
   private final Map<String, TransactionExecutor> transactionExecutors;
   private final MySqlNamedBlobDbConfig config;
+  private final Counter namedDataInconsistentGetCount;
+  private final Counter namedDataInconsistentListCount;
+  private final Counter namedDataInconsistentDeleteCount;
```

Comment:
I suggest making an inner class specifically for metrics. Something like
```
private static class Metrics {
}
```
But make sure you have MySqlNamedBlobDb to initialize the name of the metrics.

Reply:
Done in 
c7b422b